components/profile_popover/profile_popover.tsx

```diff
@@ -16,263 +13,270 @@ import {browserHistory} from 'utils/browser_history';
 import Popover from 'components/widgets/popover';
-import SharedUserIndicator from 'components/shared_user_indicator.tsx';
+import SharedUserIndicator from 'components/shared_user_indicator';
 import CustomStatusEmoji from 'components/custom_status/custom_status_emoji';
 import CustomStatusModal from 'components/custom_status/custom_status_modal';
 import CustomStatusText from 'components/custom_status/custom_status_text';
+import {UserCustomStatus, UserProfile, UserTimezone} from 'mattermost-redux/types/users';
+import {Dictionary} from 'mattermost-redux/types/utilities';
 
 import './profile_popover.scss';
 
+interface ProfilePopoverProps extends Omit<React.ComponentProps<typeof Popover>, 'id'>{
+
+    /**
+     * Source URL from the image to display in the popover
+     */
+    src: string;
+
+    /**
+     * Source URL from the image that should override default image
+     */
+    overwriteIcon?: React.ReactNode;
+
+    /**
+     * User the popover is being opened for
+     */
+    user?: Partial<UserProfile>;
+    userId: string;
+    channelId?: string;
+
+    /**
+     * Status for the user, either 'offline', 'away', 'dnd' or 'online'
+     */
+    status?: string;
+    hideStatus?: boolean;
+
+    /**
+     * Function to call to hide the popover
+     */
+    hide?: () => void;
+
+    /**
+     * Set to true if the popover was opened from the right-hand
+     * sidebar (comment thread, search results, etc.)
+     */
+    isRHS?: boolean;
+    isBusy?: boolean;
+
+    /**
+     * Returns state of modals in redux for determing which need to be closed
+     */
+    modals?: {
+        [modalId: string]: {
+            open: boolean;
+            dialogProps: Dictionary<any>;
+            dialogType: React.Component;
+        };
+    };
+    currentTeamId: string;
+
+    /**
+     * @internal
+     */
+    currentUserId: string;
+    customStatus?: UserCustomStatus | null;
+    isCustomStatusEnabled: boolean;
+
+    /**
+     * @internal
+     */
+    hasMention?: boolean;
+
+    /**
+     * @internal
+     */
+    isInCurrentTeam: boolean;
+
+    /**
+     * @internal
+     */
+    teamUrl: string;
+
+    /**
+     * @internal
+     */
+    isTeamAdmin: boolean;
+
+    /**
+     * @internal
+     */
+    isChannelAdmin: boolean;
+
+    /**
+     * @internal
+     */
+    canManageAnyChannelMembersInCurrentTeam: boolean;
+
+    /**
+     * The overwritten username that should be shown at the top of the popover
+     */
+    overwriteName?: React.ReactNode;
+
+    /**
+     * @internal
+     */
+    enableTimezone: boolean;
+    actions: {
+        openModal: (modalData: {modalId: string; dialogType: any; dialogProps?: any}) => Promise<{
+            data: boolean;
+        }>;
+        closeModal: (modalId: string) => Promise<{
+            data: boolean;
+        }>;
+        openDirectChannelToUserId: (userId?: string) => Promise<any>;
+        getMembershipForEntities: (teamId: any, userId: any, channelId: any) => Promise<any>;
```

Comment:
Most ids are `string`

Reply:
@devinbinnie what about Promise<any> where can I look for it's type

Reply:
In this case we can actually return `void` or `Promise<void>` because this component doesn't use the result of the promise.

Reply:
Done

Reply:
We can do `{error: ServerError}`

Reply:
Okay Thanks